apply/apply.go

```diff
@@ -117,10 +97,44 @@ func NewDefaultApplier(cluster *v2.Cluster) (applydriver.Interface, error) {
 		return nil, err
 	}
 
+	hostList := utils.GetIPListFromHosts(cluster.Spec.Hosts)
+
+	if err := checkAllHostsSameFamily(hostList); err != nil {
+		return nil, err
+	}
+
+	if len(hostList) > 0 && k8snet.IsIPv6String(hostList[0]) &&
+		env.ConvertEnv(cluster.Spec.Env)[v2.EnvHostIPFamily] == nil {
+		cluster.Spec.Env = append(cluster.Spec.Env, fmt.Sprintf("%s=%s", v2.EnvHostIPFamily, k8snet.IPv6))
+	}
+
 	return &applydriver.Applier{
 		ClusterDesired:      cluster,
+		ClusterFile:         file,
 		ImageManager:        imgSvc,
 		ClusterImageMounter: mounter,
 		ImageStore:          is,
 	}, nil
 }
+
+func checkAllHostsSameFamily(nodeList []string) error {
```

Comment:
This is the validate function, you could add them into validate.go . Actually It seems to have nothing to do with the main apply procedure. @VinceCui